### PR TITLE
Assign lower bound directly to likelihood variance.

### PR DESCRIPTION
**Pull request by frgsimpson**
For an easier transition from v1, and improved stability during optimisation, this sets a lower bound on the likelihood variance. 

Compared to imposing the lower bound within the positive transform, this is (a) more transparent to the user and (b) does not impose the restriction on other Parameters.

### CODE REVIEW

File: gpflow/likelihoods/likelihoods.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def variational_expectations(self, Fmu, Fvar, Y):
 class Gaussian(Likelihood):
     def __init__(self, variance=1.0, **kwargs):
         super().__init__(**kwargs)
-        self.variance = Parameter(variance, transform=positive())
+        self.variance = Parameter(variance, transform=positive(lower=1e-6))
```
Comment (awav) [inline gpflow/likelihoods/likelihoods.py]: This line deserves an explanation in documentation.

**Comment by st--** [comment]
@frgsimpson Right now, this change would be a no-op, as the default lower for the positive transform is also 1e-6. Did you mean to then change that to None or 1e-10 or something else smaller ?

**Comment by jahall** [comment]
> @frgsimpson Right now, this change would be a no-op, as the default lower for the positive transform is also 1e-6. Did you mean to then change that to None or 1e-10 or something else smaller ?

Actually the default is still `None` as per the thread in #1148

**Comment by awav** [comment]
@st-- ,

> Right now, this change would be a no-op

.The current default value for the positive shift is None and therefore no shift is applied. Default value was discussed with @frgsimpson, @markvdw. The obscurity of the fact that all positive values by default can't go below `1e-6` has big impact on results misinterpretation for a model. The point in favor of having `1e-6` threshold is that it **might** have positive effect on optimization. What if it is not and moreover we don't see much of the change in current tests for example. Now, we have different settings in the config for changing the lower positive value minimum and the type of positive transformation - setting 'right' (e.g. `1e-10`, `1e-6` and etc.) default values doesn't look right for this degree of freedom.

File: gpflow/likelihoods/likelihoods.py (hunk 1 above)
Comment (awav) [reply gpflow/likelihoods/likelihoods.py]: @frgsimpson ^^^

Comment (frgsimpson) [reply gpflow/likelihoods/likelihoods.py]: @awav I've just added a docstring for the class, let me know if that's what you had in mind

Comment (st--) [reply gpflow/likelihoods/likelihoods.py]: Actually, how about just turning this into a keyword argument to the class ?
```python
    def __init__(self, variance=1.0, variance_lower_bound=1e-6, **kwargs):
        super().__init__(**kwargs)
        self.variance = Parameter(variance, transform=positive(lower=variance_lower_bound))
```

Comment (frgsimpson) [reply gpflow/likelihoods/likelihoods.py]: Done!

**Pull request merged**